### ISSUE TITLE
Installing Gemfile dependencies on Windows fails

### ISSUE THREAD

**Issue by dan-wittenberg** [puppetlabs/pdk#297]
I did a clean install of PDK 1.1.0.0 yesterday, Windows 10, and trying to run pdk validate in a module:

```
PS C:\Users\dwitt\Documents\git\ahead> pdk validate
pdk (INFO): Running all available validators...
[X] Installing missing Gemfile dependencies.
pdk (FATAL): Fetching https://github.com/bobtfish/hiera-puppet-helper.git

Retrying `git clone --no-checkout --quiet "C:/Users/dwitt/AppData/Local/PDK/cache/ruby/2.1.0/cache/bundler/git/hiera-puppet-helper-03d5c001b37179971b
a1a942818bb895676603a0" "C:/Users/dwitt/AppData/Local/PDK/cache/ruby/2.1.0/bundler/gems/hiera-puppet-helper-155f132c0b22"` due to error (2/4): Bundle
r::Source::Git::GitCommandError Git error: command `git clone --no-checkout --quiet "C:/Users/dwitt/AppData/Local/PDK/cache/ruby/2.1.0/cache/bundler/
git/hiera-puppet-helper-03d5c001b37179971ba1a942818bb895676603a0" "C:/Users/dwitt/AppData/Local/PDK/cache/ruby/2.1.0/bundler/gems/hiera-puppet-helper
-155f132c0b22"` in directory C:/Users/dwitt/Documents/git/ahead has failed.
If this error persists you could try removing the cache directory 'C:/Users/dwitt/AppData/Local/PDK/cache/ruby/2.1.0/cache/bundler/git/hiera-puppet-h
elper-03d5c001b37179971ba1a942818bb895676603a0'
Retrying `git clone --no-checkout --quiet "C:/Users/dwitt/AppData/Local/PDK/cache/ruby/2.1.0/cache/bundler/git/hiera-puppet-helper-03d5c001b37179971b
a1a942818bb895676603a0" "C:/Users/dwitt/AppData/Local/PDK/cache/ruby/2.1.0/bundler/gems/hiera-puppet-helper-155f132c0b22"` due to error (3/4): Bundle
r::Source::Git::GitCommandError Git error: command `git clone --no-checkout --quiet "C:/Users/dwitt/AppData/Local/PDK/cache/ruby/2.1.0/cache/bundler/
git/hiera-puppet-helper-03d5c001b37179971ba1a942818bb895676603a0" "C:/Users/dwitt/AppData/Local/PDK/cache/ruby/2.1.0/bundler/gems/hiera-puppet-helper
-155f132c0b22"` in directory C:/Users/dwitt/Documents/git/ahead has failed.
If this error persists you could try removing the cache directory 'C:/Users/dwitt/AppData/Local/PDK/cache/ruby/2.1.0/cache/bundler/git/hiera-puppet-h
elper-03d5c001b37179971ba1a942818bb895676603a0'
Retrying `git clone --no-checkout --quiet "C:/Users/dwitt/AppData/Local/PDK/cache/ruby/2.1.0/cache/bundler/git/hiera-puppet-helper-03d5c001b37179971b
a1a942818bb895676603a0" "C:/Users/dwitt/AppData/Local/PDK/cache/ruby/2.1.0/bundler/gems/hiera-puppet-helper-155f132c0b22"` due to error (4/4): Bundle
r::Source::Git::GitCommandError Git error: command `git clone --no-checkout --quiet "C:/Users/dwitt/AppData/Local/PDK/cache/ruby/2.1.0/cache/bundler/
git/hiera-puppet-helper-03d5c001b37179971ba1a942818bb895676603a0" "C:/Users/dwitt/AppData/Local/PDK/cache/ruby/2.1.0/bundler/gems/hiera-puppet-helper
-155f132c0b22"` in directory C:/Users/dwitt/Documents/git/ahead has failed.
If this error persists you could try removing the cache directory 'C:/Users/dwitt/AppData/Local/PDK/cache/ruby/2.1.0/cache/bundler/git/hiera-puppet-h
elper-03d5c001b37179971ba1a942818bb895676603a0'
Git error: command `git clone --no-checkout --quiet
"C:/Users/dwitt/AppData/Local/PDK/cache/ruby/2.1.0/cache/bundler/git/hiera-puppet-helper-03d5c001b37179971ba1a942818bb895676603a0"
"C:/Users/dwitt/AppData/Local/PDK/cache/ruby/2.1.0/bundler/gems/hiera-puppet-helper-155f132c0b22"`
in directory C:/Users/dwitt/Documents/git/ahead has failed.
If this error persists you could try removing the cache directory
'C:/Users/dwitt/AppData/Local/PDK/cache/ruby/2.1.0/cache/bundler/git/hiera-puppet-helper-03d5c001b37179971ba1a942818bb895676603a0'

error: cannot spawn sh: No such file or directory
fatal: unable to fork
error: cannot spawn sh: No such file or directory
fatal: unable to fork
error: cannot spawn sh: No such file or directory
fatal: unable to fork
error: cannot spawn sh: No such file or directory
fatal: unable to fork

pdk (FATAL): Unable to install missing Gemfile dependencies.
```

**Comment by james-stocks** [comment]
Hello -
Can you provide the output for `pdk validate --debug` please?

**Comment by dan-wittenberg** [comment]
PS C:\Users\dwitt\Documents\git\ahead> pdk validate --debug
pdk (INFO): Running all available validators...
pdk (DEBUG): Checking for missing Gemfile dependencies.
pdk (DEBUG): Using 'private/ruby/2.1.9/bin/bundle.bat' from PDK package.
pdk (DEBUG): Executing 'C:/PROGRA~1/PUPPET~1/DevelopmentKit/private/ruby/2.1.9/bin/bundle.bat check --gemfile=C:/Users/dwitt/Documents/git/ahead/Gemf
ile'
pdk (DEBUG): Command environment: GEM_HOME is 'C:\Users\dwitt\AppData\Local/PDK/cache/ruby/2.1.0' and GEM_PATH is 'C:/PROGRA~1/PUPPET~1/DevelopmentKi
t/share/cache/ruby/2.1.0'
pdk (DEBUG): Execution of 'C:/PROGRA~1/PUPPET~1/DevelopmentKit/private/ruby/2.1.9/bin/bundle.bat check --gemfile=C:/Users/dwitt/Documents/git/ahead/G
emfile' complete (duration: 1.109287s; exit code: 11)
pdk (DEBUG): https://github.com/bobtfish/hiera-puppet-helper.git (at master@155f132) is not
yet checked out. Run `bundle install` first.


pdk (DEBUG): Using 'private/ruby/2.1.9/bin/bundle.bat' from PDK package.
pdk (DEBUG): Executing 'C:/PROGRA~1/PUPPET~1/DevelopmentKit/private/ruby/2.1.9/bin/bundle.bat install --gemfile=C:/Users/dwitt/Documents/git/ahead/Ge
mfile -j4'
pdk (DEBUG): Command environment: GEM_HOME is 'C:\Users\dwitt\AppData\Local/PDK/cache/ruby/2.1.0' and GEM_PATH is 'C:/PROGRA~1/PUPPET~1/DevelopmentKi
t/share/cache/ruby/2.1.0'
pdk (DEBUG): Execution of 'C:/PROGRA~1/PUPPET~1/DevelopmentKit/private/ruby/2.1.9/bin/bundle.bat install --gemfile=C:/Users/dwitt/Documents/git/ahead
/Gemfile -j4' complete (duration: 1.636631s; exit code: 11)
pdk (FATAL):
Retrying `git clone --no-checkout --quiet "C:/Users/dwitt/AppData/Local/PDK/cache/ruby/2.1.0/cache/bundler/git/hiera-puppet-helper-03d5c001b37179971b
a1a942818bb895676603a0" "C:/Users/dwitt/AppData/Local/PDK/cache/ruby/2.1.0/bundler/gems/hiera-puppet-helper-155f132c0b22"` due to error (2/4): Bundle
r::Source::Git::GitCommandError Git error: command `git clone --no-checkout --quiet "C:/Users/dwitt/AppData/Local/PDK/cache/ruby/2.1.0/cache/bundler/
git/hiera-puppet-helper-03d5c001b37179971ba1a942818bb895676603a0" "C:/Users/dwitt/AppData/Local/PDK/cache/ruby/2.1.0/bundler/gems/hiera-puppet-helper
-155f132c0b22"` in directory C:/Users/dwitt/Documents/git/ahead has failed.
If this error persists you could try removing the cache directory 'C:/Users/dwitt/AppData/Local/PDK/cache/ruby/2.1.0/cache/bundler/git/hiera-puppet-h
elper-03d5c001b37179971ba1a942818bb895676603a0'
Retrying `git clone --no-checkout --quiet "C:/Users/dwitt/AppData/Local/PDK/cache/ruby/2.1.0/cache/bundler/git/hiera-puppet-helper-03d5c001b37179971b
a1a942818bb895676603a0" "C:/Users/dwitt/AppData/Local/PDK/cache/ruby/2.1.0/bundler/gems/hiera-puppet-helper-155f132c0b22"` due to error (3/4): Bundle
r::Source::Git::GitCommandError Git error: command `git clone --no-checkout --quiet "C:/Users/dwitt/AppData/Local/PDK/cache/ruby/2.1.0/cache/bundler/
git/hiera-puppet-helper-03d5c001b37179971ba1a942818bb895676603a0" "C:/Users/dwitt/AppData/Local/PDK/cache/ruby/2.1.0/bundler/gems/hiera-puppet-helper
-155f132c0b22"` in directory C:/Users/dwitt/Documents/git/ahead has failed.
If this error persists you could try removing the cache directory 'C:/Users/dwitt/AppData/Local/PDK/cache/ruby/2.1.0/cache/bundler/git/hiera-puppet-h
elper-03d5c001b37179971ba1a942818bb895676603a0'
Retrying `git clone --no-checkout --quiet "C:/Users/dwitt/AppData/Local/PDK/cache/ruby/2.1.0/cache/bundler/git/hiera-puppet-helper-03d5c001b37179971b
a1a942818bb895676603a0" "C:/Users/dwitt/AppData/Local/PDK/cache/ruby/2.1.0/bundler/gems/hiera-puppet-helper-155f132c0b22"` due to error (4/4): Bundle
r::Source::Git::GitCommandError Git error: command `git clone --no-checkout --quiet "C:/Users/dwitt/AppData/Local/PDK/cache/ruby/2.1.0/cache/bundler/
git/hiera-puppet-helper-03d5c001b37179971ba1a942818bb895676603a0" "C:/Users/dwitt/AppData/Local/PDK/cache/ruby/2.1.0/bundler/gems/hiera-puppet-helper
-155f132c0b22"` in directory C:/Users/dwitt/Documents/git/ahead has failed.
If this error persists you could try removing the cache directory 'C:/Users/dwitt/AppData/Local/PDK/cache/ruby/2.1.0/cache/bundler/git/hiera-puppet-h
elper-03d5c001b37179971ba1a942818bb895676603a0'
Git error: command `git clone --no-checkout --quiet
"C:/Users/dwitt/AppData/Local/PDK/cache/ruby/2.1.0/cache/bundler/git/hiera-puppet-helper-03d5c001b37179971ba1a942818bb895676603a0"
"C:/Users/dwitt/AppData/Local/PDK/cache/ruby/2.1.0/bundler/gems/hiera-puppet-helper-155f132c0b22"`
in directory C:/Users/dwitt/Documents/git/ahead has failed.
If this error persists you could try removing the cache directory
'C:/Users/dwitt/AppData/Local/PDK/cache/ruby/2.1.0/cache/bundler/git/hiera-puppet-helper-03d5c001b37179971ba1a942818bb895676603a0'

error: cannot spawn sh: No such file or directory
fatal: unable to fork
error: cannot spawn sh: No such file or directory
fatal: unable to fork
error: cannot spawn sh: No such file or directory
fatal: unable to fork
error: cannot spawn sh: No such file or directory
fatal: unable to fork

pdk (FATAL): Unable to install missing Gemfile dependencies.
pdk (DEBUG): C:/PROGRA~1/PUPPET~1/DEVELO~1/private/ruby/2.1.9/lib/ruby/gems/2.1.0/gems/pdk-1.1.0/lib/pdk/util/bundler.rb:44:in `ensure_bundle!'
pdk (DEBUG): C:/PROGRA~1/PUPPET~1/DEVELO~1/private/ruby/2.1.9/lib/ruby/gems/2.1.0/gems/pdk-1.1.0/lib/pdk/cli/validate.rb:81:in `block (2 levels) in <
module:CLI>'
pdk (DEBUG): C:/PROGRA~1/PUPPET~1/DEVELO~1/private/ruby/2.1.9/lib/ruby/gems/2.1.0/gems/cri-2.9.1/lib/cri/command.rb:319:in `call'
pdk (DEBUG): C:/PROGRA~1/PUPPET~1/DEVELO~1/private/ruby/2.1.9/lib/ruby/gems/2.1.0/gems/cri-2.9.1/lib/cri/command.rb:319:in `run_this'
pdk (DEBUG): C:/PROGRA~1/PUPPET~1/DEVELO~1/private/ruby/2.1.9/lib/ruby/gems/2.1.0/gems/cri-2.9.1/lib/cri/command.rb:264:in `run'
pdk (DEBUG): C:/PROGRA~1/PUPPET~1/DEVELO~1/private/ruby/2.1.9/lib/ruby/gems/2.1.0/gems/cri-2.9.1/lib/cri/command.rb:282:in `run'
pdk (DEBUG): C:/PROGRA~1/PUPPET~1/DEVELO~1/private/ruby/2.1.9/lib/ruby/gems/2.1.0/gems/pdk-1.1.0/lib/pdk/cli.rb:17:in `run'
pdk (DEBUG): C:/PROGRA~1/PUPPET~1/DEVELO~1/private/ruby/2.1.9/lib/ruby/gems/2.1.0/gems/pdk-1.1.0/exe/pdk:6:in `<top (required)>'
pdk (DEBUG): C:/PROGRA~1/PUPPET~1/DEVELO~1/private/ruby/2.1.9/bin/pdk:23:in `load'
pdk (DEBUG): C:/PROGRA~1/PUPPET~1/DEVELO~1/private/ruby/2.1.9/bin/pdk:23:in `<main>'

**Comment by rodjek** [comment]
@dan-wittenberg I've created a [Jira ticket](https://tickets.puppetlabs.com/browse/PDK-502) to track this

**Comment by rodjek** [comment]
This has been resolved in #298 and will be part of the next release.
